src/pages/base.js

```diff
@@ -14,17 +14,27 @@ class Page {
   }
 
   visitWithParameters(data) {
-    let url = this.url;
-
-    for (const item of data.raw()) {
-      url = url.replace(`:${item[0]}`, item[1]);
-    }
-
-    if (config.type === 'otherWeb') {
-      return protractor.browser.get(url);
-    }
-
-    return protractor.browser.get(url).then(() => protractor.browser.waitForAngular());
+    const esc = encodeURIComponent;
+    const additionalParams = [];
+
+    this.url =
```

Comment:
If I'm correct, this won't work correctly when we call this page object with a different set of "path" parameters.
Url variable is a placeholder, so we should rather define a template for it and then build the url to be accessed, than  mutate the template itself.

Another point is that by current implementation we don't know if a param is a query or path parameter. I think this is a good moment to extend step arguments to 3 columns where one would define a type of param (query or path).

Reply:
And I don't the other part is needed. It is clear that additional params are those that were not included in url pattern.